include/nanogui/colorpicker.h

```diff
@@ -21,24 +22,84 @@ NAMESPACE_BEGIN(nanogui)
 
-    /// Set the change callback
-    std::function<void(const Color &)> callback() const                  { return mCallback; }
+    /**
+     * The callback to execute when a new Color is selected on the ColorWheel
+     * **and** the user clicks the \ref nanogui::ColorPicker::mPickButton.
+     */
+    std::function<void(const Color &)> callback() const { return mCallback; }
+
+    /**
+     * The callback to execute when a new Color is selected on the ColorWheel
+     * **and** the user clicks the \ref nanogui::ColorPicker::mPickButton.
+     */
     void setCallback(const std::function<void(const Color &)> &callback) { mCallback = callback; }
 
-    /// Get the current color
+    /// Get the current Color selected for this ColorPicker.
     Color color() const;
-    /// Set the current color
+
+    /// Set the current Color selected for this ColorPicker.
     void setColor(const Color& color);
+
+    /// The current caption of the \ref nanogui::ColorPicker::mPickButton.
+    const std::string &pickButtonCaption() { return mPickButton->caption(); }
+
+    /// Sets the current caption of the \ref nanogui::ColorPicker::mPickButton.
+    void setPickButtonCaption(const std::string &caption) { mPickButton->setCaption(caption); }
+
+    /// The current caption of the \ref nanogui::ColorPicker::mResetButton.
+    const std::string &resetButtonCaption() { return mResetButton->caption(); }
+
+    /// Sets the current caption of the \ref nanogui::ColorPicker::mResetButton.
+    void setResetButtonCaption(const std::string &caption) { mResetButton->setCaption(caption); }
+
 protected:
+    /**
+     * The callback to execute when a new Color is selected on the ColorWheel
+     * **and** the user clicks the \ref nanogui::ColorPicker::mPickButton.
+     */
     std::function<void(const Color &)> mCallback;
+
+    /// The ColorWheel for this ColorPicker (the actual widget allowing selection).
     ColorWheel *mColorWheel;
+
+    /**
+     * The Button used to signal that the current value on the ColorWheel is the
+     * desired color to be chosen.  The default value for the caption of this
+     * Button is ``"Pick"``.  You can change it using
+     * \ref nanogui::ColorPicker::setPickButtonCaption if you need.
+     *
+     * The color of this Button will not affect \ref nanogui::ColorPicker::color
+     * until the user has actively selected by clicking this pick button.
+     * Similarly, the \ref nanogui::ColorPicker::mCallback function is only
+     * called when a user selects a new Color using by clicking this Button.
+     */
     Button *mPickButton;
+
+    /**
+     * Remains the Color of the active color selection, until the user picks a
+     * new Color on the ColorWheel **and** selects the
+     * \ref nanogui::ColorPicker::mPickButton.  The default value for the
+     * caption of this Button is ``"Reset"``.  You can change it using
+     * \ref nanogui::ColorPicker::setResetButtonCaption if you need.
+     */
+    Button *mResetButton;
```

Comment:
This is new.  The changed functionality of ColorPicker necessitates this.  See comments in cpp file.